### ISSUE TITLE
WebApp authentication flow without secret key 

### ISSUE THREAD

**Issue by alexagr** [AzureAD/microsoft-authentication-library-for-python#338]
I want to add authentication via Azure AD to my Python + Flask Web App and [quickstart](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-v2-python-webapp#how-the-sample-works) works great. However it requires me to add secret key to my application and provision it in the WebApp.

As far as I understand from [Authorization Code Flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow) documentation, if I'm not calling any Azure API from my WebApp it should be possible to implement authentication ONLY without any secret key. Essentially I should be able to exchange authorization code for id_token - which would be enough to perform user authentication.

Is there any chance that you could add this authentication flow to MSAL library?

**Comment by rayluo** [comment]
In that quickstart, you can try set the `CLIENT_SECRET` as `None` in `app_config.py`. Let us know how it pans out.

**Comment by alexagr** [comment]
If I do that, I get the following error from **acquire_token_by_auth_code_flow()**: 
AADSTS7000218: The request body must contain the following parameter: 'client_assertion' or 'client_secret'.

**Comment by alexagr** [comment]
I think access to /token endpoint (to exchange authorization code for tokens) always requires client_secret.
So the correct way to implement "pure authentication" flow is to request id_token instead of authorization code from /authorize endpoint - as described [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc) - and then validate identity token in the redirect_url callback.

**Comment by rayluo** [comment]
I tried it just now. It turns out, in the Azure Portal, your app's authentication "Platform configurations" should configure the redirect_uri inside the "Mobile and desktop applications" platform, rather than in "Web" platform. The latter implies that your app would need to provide a secret (or cert, for that matter).

**Comment by alexagr** [comment]
Right. My point is that if my Web app doesn't need to access Azure services there is no need for it to have credentials at all. This is backed up "Integration Assistant" - if you choose Web app and specify that the application is not calling APIs it will report configured credentials as "Discouraged configuration".
In any case, I think that I figured it out on my own - I simply use OpenID Flow and acquire identity token directly from `/authorize` endpoint - and this works just fine. I packed the code it into a simple library and published it on Github: https://github.com/alexagr/simple-azure-auth .
